spec/models/email_authentication_spec.rb

```diff
@@ -1,47 +1,89 @@
 require "rails_helper"
 
 RSpec.describe EmailAuthentication, type: :model do
-  let(:weak_password_error) { "Please choose a stronger password with at least 8 characters. Try a mix of letters, numbers, and symbols." }
-
   describe "Associations" do
     it { should have_one(:user_authentication) }
     it { should have_one(:user).through(:user_authentication) }
   end
 
-  it "requires a non blank password" do
-    auth = build(:email_authentication, password: "")
-    expect(auth).to_not be_valid
-    expect(auth.errors.messages[:password]).to eq [weak_password_error]
-  end
+  describe "password" do
+    it "can't be nil" do
+      auth = build(:email_authentication, password: nil)
+      expect(auth).to_not be_valid
+      # expect(auth.errors.messages[:password]).to include "can't be blank"
```

Comment:
All of these commented out error message tests are intentional. They will be made live when the UI is changed to handle multiple error messages.

Reply:
I'm not sure I follow -- wouldn't these model validations pass right now, because you have added the new validations above.  How does this depend upon the UI?

Reply:
Currently, the validation messages get wiped out (if there's more than one of them) by this `after_validation` in the EmailAuthentication model.
```
  # We only want to display one error message to the user, so if we get multiple
  # errors clear out all errors and present our nice message to the user.
  def strip_unnecessary_errors
    if errors[:password].any? && errors[:password].size > 1
      errors.delete(:password)
      errors.add(:password, I18n.translate("errors.messages.password.password_strength"))
    end
  end
  ```
If you turn that validation off and have multiple error messages, you get this:
<img width="590" alt="image" src="https://user-images.githubusercontent.com/28661253/108275765-cf9fc000-7144-11eb-8ee6-4856c001cc06.png">
Not terrible but not ideal. I'm sure I could fix that in the view, but I figured I would leave it alone for now. I want to have multiple error messages to enable the password requirements checklist. So my plan was to just leave the error message overwriting until I have the UI changes for handling multiple errors.

Reply:
I think removing that `after_validation` is fine, because the `nice message` it presents won't match these new validation rules: `Please choose a stronger password with at least 8 characters. Try a mix of letters, numbers, and symbols."`. Plus its better to make sure these specs are really running when they get merged, just to be sure our password rules are all good.

If we have ugly (but accurate) error messages displayed to a new user until you have the UI piece shipped, that seems fine to me.

Reply:
Works for me. I did update that "nice message" to be correct, but I don't think that changes anything. It seems the more important point is perhaps that we can live with an ugly error message on a low-use page for a few days.

Reply:
Works for me. I did update that "nice message" to be correct, but I don't think that changes anything. It seems the more important point is perhaps that we can live with an ugly error message on a low-use page for a few days.